### PR TITLE
feat(incident): add workflows execution tab to incident

### PR DESCRIPTION
**Pull request by shahargl**
close https://github.com/keephq/keep/issues/2047
close https://github.com/keephq/keep/issues/2045

### CODE REVIEW

**Review by VladimirFilonov — APPROVED** [review]
Love it!

**Review by VladimirFilonov — DISMISSED** [review]

File: keep-ui/app/incidents/[id]/incident-workflow-sidebar.tsx [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+
+  useEffect(() => {
+    if (isOpen && selectedExecution) {
+      // You can add any side effects here when the sidebar opens
+    }
+  }, [isOpen, selectedExecution]);
+
+  if (!selectedExecution) return null;
+
+  const customFormatter = (value: number, unit: string, suffix: string) => {
```
Comment (Kiryous) [inline keep-ui/app/incidents/[id]/incident-workflow-sidebar.tsx:38]: This is unused, could be removed or is it intended to be used somewhere?

File: keep-ui/app/incidents/[id]/incident-workflow-sidebar.tsx [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+  toggle,
+  selectedExecution,
+}) => {
+  const { data: workflowExecutionData } = useWorkflowExecution(
+    selectedExecution?.workflow_id || "",
+    selectedExecution?.id || ""
+  );
+
+  useEffect(() => {
+    if (isOpen && selectedExecution) {
```
Comment (Kiryous) [inline keep-ui/app/incidents/[id]/incident-workflow-sidebar.tsx:31]: Is it intended? If this is a comment for future, let's comment the whole useEffect to not to run it idle

File: keep-ui/app/incidents/[id]/incident-workflow-sidebar.tsx [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+          leave="transition ease-in-out duration-300 transform"
+          leaveFrom="translate-x-0"
+          leaveTo="translate-x-full"
+        >
+          <Dialog.Panel className="fixed right-0 inset-y-0 w-2/4 bg-white z-30 p-6 overflow-auto flex flex-col">
+            <div className="flex justify-between mb-4">
+              <div>
+                <Dialog.Title className="text-3xl font-bold" as={Title}>
+                  Workflow Execution Details
+                  <Badge className="ml-4" color="orange">
```
Comment (Kiryous) [inline keep-ui/app/incidents/[id]/incident-workflow-sidebar.tsx:83]: let's capitalize the status, which is now all in lowercase and pick color like icon, depending on the status value
```suggestion
                  <Badge className="ml-4 capitalize" color={selectedExecution.status === "error" ? "red" : selectedExecution.status === "success" ? "green" : "orange"}>
```

File: keep-ui/app/incidents/[id]/incident-workflow-sidebar.tsx [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+                    </Text>
+                    <TextInput value={selectedExecution.id} readOnly />
+                  </div>
+                  <div>
+                    <Text className="block text-sm font-medium text-gray-700 mb-2">
+                      Status
+                    </Text>
+                    <div className="flex items-center">
+                      {getIcon(selectedExecution.status)}
+                      <span className="ml-2">{selectedExecution.status}</span>
```
Comment (Kiryous) [inline keep-ui/app/incidents/[id]/incident-workflow-sidebar.tsx:110]: let's capitalize the status
```suggestion
                      <span className="ml-2 capitalize">{selectedExecution.status}</span>
```

**Review by Kiryous — CHANGES_REQUESTED** [review]
awesome work! added a few minor comments on UI

**Review by Kiryous — APPROVED** [review]
LGTM!

**Pull request merged**